tests/interop/messages/cancel_test.go

```diff
@@ -0,0 +1,45 @@
+package messages_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/xconnio/wampproto-go/messages"
+	"github.com/xconnio/wampproto-go/serializers"
+	"github.com/xconnio/wampproto-go/tests"
+)
+
+func cancelsEqual(msg1 *messages.Cancel, msg2 *messages.Cancel) bool {
+	return msg1.RequestID() == msg2.RequestID() &&
+		reflect.DeepEqual(msg1.Options(), msg2.Options())
+}
+
+func TestCancelJSONSerializer(t *testing.T) {
+	var message = messages.NewCancel(1, map[string]any{})
```

Comment:
These should be a single test with 3 sub tests for each serializer. We also need to make sure to reuse code here instead of repeating